src/pspm_select_channels.m

```diff
@@ -45,12 +52,15 @@
 %% 3 Select channels 
 channeltype_list = cellfun(@(x) x.header.chantype, data, 'uni', false);
 channelunits_list = cellfun(@(x) x.header.units, data, 'uni', false);
-if ischar(channel)  && contains(channel, 'event')
-    pos_of_channels = find(strcmpi(channelunits_list, 'events'));
+if strcmp(units, 'any')
+    units = channelunits_list;
+end
+if ischar(channel) && contains(channel, 'event')
+    pos_of_channels = find(strcmpi(channelunits_list, 'events') & strcmp(channelunits_list, units));
 elseif ischar(channel) && strcmpi(channel, 'wave')
-    pos_of_channels = find(~strcmpi(channelunits_list, 'events'));
+    pos_of_channels = find(~strcmpi(channelunits_list, 'events') & strcmp(channelunits_list, units));
 elseif ischar(channel) 
-    pos_of_channels = find(contains(channeltype_list, channel));
+    pos_of_channels = find(contains(channeltype_list, channel) & strcmp(channelunits_list, units));
```

Comment:
I saw the logic and I think `pos_of_channel` should be a numeric array such as `[5, 8, 12]`? If we use `&`, it will become `1` or `0`, for example `[5,8,12]&[1,1,0]` will give `[1,1,0]`. I dont think this is what that is expected?

Reply:
The syntax is `find(... & ...)` where `...` are logical expressions, which will return a numerical index.